server/src/com/cloud/ha/HighAvailabilityManagerImpl.java

```diff
@@ -282,7 +282,9 @@ public void scheduleRestartForVmsOnHost(final HostVO host, boolean investigate)
                         + hostId + " VM HA is done");
                 continue;
             }
-            scheduleRestart(vm, investigate);
+            if (vm.isHaEnabled()) {
```

Comment:
Why not skip the loop with a `continue` around line 268 when vm is not HA enabled?